server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -316,6 +316,17 @@ public void consume(QuerySearchResult result, Runnable next) {
                         emptyResults.add(new SearchShard(target.getClusterAlias(), target.getShardId()));
                     }
                 } else {
+                    if (hasAggs) {
```

Comment:
May be not related to this PR, I wonder why we limit circuit breaker exceptions only to aggs? What if we have huge docs that need to be merged?

Reply:
The content of the document is retrieved during the fetch phase so that's a different issue. Here we're only dealing with shard aggregation results and `TopDocs`.